assertj-core/src/main/java/org/assertj/core/api/DefaultAssertionErrorCollector.java

```diff
@@ -74,6 +75,22 @@ public List<AssertionError> assertionErrorsCollected() {
     return decorateErrorsCollected(errors);
   }
 
+  /**
+   * Same as {@link DefaultAssertionErrorCollector#addAfterAssertionErrorCollected(AfterAssertionErrorCollected)}, but
+   * also removes all previously added callbacks. Please consider using
+   * {@link DefaultAssertionErrorCollector#addAfterAssertionErrorCollected(AfterAssertionErrorCollected)}
+   * instead, because another frameworks and integrations can also use this functionality.
+   *
+   * @param afterAssertionErrorCollected the callback.
+   *
+   * @since 3.17.0
+   * @deprecated
```

Comment:
I don't see it as deprecated, users that already register only one callback don't want to migrate to `addAfterAssertionErrorCollected` so we should keep the initial javadoc just adding that it clears all callbacks previously added with `addAfterAssertionErrorCollected` (as you did)

Reply:
added to commit